target/linux/mpc85xx/files/arch/powerpc/boot/dts/hiveap-370.dts

```diff
@@ -0,0 +1,295 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/include/ "fsl/p1020si-pre.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/leds/common.h>
+/ {
+	model = "Aerohive HiveAP-370";
+	compatible = "aerohive,hiveap-370";
+
+	aliases {
+		led-boot = &led_status;
+		led-failsafe = &led_status;
+		led-running = &led_status;
+		led-upgrade = &led_status;
+		label-mac-device = &enet0;
+	};
+	
+	leds {
+		compatible = "gpio-leds";
+		led_status: status {
+			label = "ap370:white:status";
+			gpios = <&gpio0 11 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+			linux,default-trigger = "default-on";
+		};
+	};
+
+	chosen {
+		bootargs-override = "console=ttyS0,9600";
+	};
+
+	memory {
+		device_type = "memory";
+	};
+
+	board_lbc: lbc: localbus@ffe05000 {
+		reg = <0x0 0xffe05000 0x0 0x1000>;
+		ranges = <0x0 0x0 0x0 0xffa00000 0x100000>;
+		
+		flash@0,0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x1>;
+					compatible = "fsl,elbc-fcm-nand";
+					reg = <0x0 0x0 0x4000000>;
+					bank-width = <0x2>;
+					device-width = <0x1>;
+					nand-ecc-mode = "none";
+					
+					partition@0 {
+							reg = <0x0 0x20000>;
+							label = "0Uboot";
+					};
+					
+					partition@20000 {
+							reg = <0x20000 0x200000>;
+							label = "Uboot";
+					};
+					
+					partition@220000 {
+							reg = <0x220000 0x200000>;
+							label = "Uboot-backup";
+					};
+					
+					partition@420000 {
+							reg = <0x420000 0x40000>;
+							label = "Uboot Env";
+					};
+					
+					partition@460000 {
+							reg = <0x460000 0x40000>;
+							label = "Boot Info";
+					};
+					
+					partition@4a0000 {
+							reg = <0x4a0000 0x40000>;
+							label = "Static Boot Info";
+					};
+					
+					hwinfo: partition@4e0000 {
+							reg = <0x4e0000 0x40000>;
+							label = "Hardware Info";
+					};
+					
+					partition@520000 {
```

Comment:
In order to allow the `rootfs` and `rootfs_data` to go into UBI, I found a better solution was to make use of the existing `nand.sh`'s behavior (by using a `sysupgrade-tar`):

`append-dtb | pad-to 256k | append-kernel | sysupgrade-tar kernel=$$$$@ | append-metadata`

Reply:
I ended up using this solution to shim the dtb before the beginning of the kernel.

This is needed because the bootloader does not accept FIT images, and cannot find the dtb if it is appended to the kernel: in this case, we instruct u-boot to load the dtb and kernel from fixed addresses, and boot from them directly.

This solution lets us also load rootfs and rootfs_data from, and sysupgrade to, UBI partitions, despite this board's u-boot's lack of UBI support.